scripts/ci_install_dependency.sh

```diff
@@ -10,7 +10,7 @@ pip install --upgrade pip
 pip install -e "python[all]" --find-links https://flashinfer.ai/whl/cu121/torch2.4/flashinfer/
 
 # Force reinstall flashinfer
-pip install flashinfer -i ${FLASHINFER_REPO} --force-reinstall
+pip install flashinfer --find-links ${FLASHINFER_REPO} --force-reinstall --no-deps
```

Comment:
There are two scenarios: one is a stable release, and the other is a nightly release. The first scenario uses cache because the script defaults to installing the stable version, making it very fast. For the second scenario, it depends on whether it's the first installation or not. It only triggers when users manually select a nightly release during PR testing, and if it's not the first time, it also uses the nightly cache, which is also very fast.